src/internal/connector/onedrive/collections_test.go

```diff
@@ -1222,7 +1328,41 @@ func (suite *OneDriveCollectionsSuite) TestGet() {
 					"folder": folderPath,
 				},
 			},
-			expectedDelList: map[string]struct{}{},
+			expectedDelList: map[string]struct{}{
+				"file": {},
+			},
+		},
+		{
+			name:   "OneDrive_FolderRenameWithinDelta_NoErrors",
+			drives: []models.Driveable{drive1},
+			items: map[string][]deltaPagerResult{
+				driveID1: {
+					{
+						items: []models.DriveItemable{
+							driveItem("folder", "folder", testBaseDrivePath, false, true, false),
+							driveItem("file", "file", testBaseDrivePath+"/folder", true, false, false),
+							driveItem("folder", "folder2", testBaseDrivePath, false, true, false),
+						},
+						deltaLink: &delta,
+					},
+				},
+			},
+			errCheck: assert.NoError,
+			expectedCollections: map[string][]string{
+				folderPath:     {"file"},
```

Comment:
with the current code indexing on folder paths I think this is actually incorrect. I think the key we'd need would be testBaseDrivePath+"/folder2" since this test renames the folder

See my other comment about the PR in general though as I'm not sure right now if indexing by collection name is a good idea